### PR TITLE
feat: allow enableIf to accept a funciton

### LINKED ISSUE #875: Allow `enableIf` to receive a function that returns a plugin
Allow something like:

```
enableIf(someBoolean, () => useSomething());
```

> This is needed in order to have plugin lazy loaded, only if the boolean is `true` and avoid running the init flow of that plugin.

### PR DESCRIPTION
**Pull request by saihaj**
Closes #875

### CODE REVIEW

**Comment by saihaj** [comment]
@dotansimha I am not sure how to test if a plugin's init is called or not : )?

File: packages/core/src/enable-if.ts [hunk 1]
```diff
@@ -16,6 +16,10 @@ export function isPluginEnabled(t: PluginOrDisabledPlugin): t is Plugin {
 /**
  * Utility function to enable a plugin.
  */
-export function enableIf(condition: boolean, plugin: Plugin): PluginOrDisabledPlugin {
-  return condition ? plugin : EnableIfBranded.DisabledPlugin;
+export function enableIf(condition: boolean, plugin: Plugin | (() => Plugin)): PluginOrDisabledPlugin {
+  if (condition) {
+    return plugin instanceof Function ? plugin() : plugin;
```
Comment (dotansimha) [inline packages/core/src/enable-if.ts]: Please replace `instanceof Function` with `typeof`, it should be faster :)

**Review by dotansimha — CHANGES_REQUESTED** [review]

**Comment by dotansimha** [comment]
> @dotansimha I am not sure how to test if a plugin's init is called or not : )?

You can create a "fake plugin" that is only a `jest.fn().mockImplementation(() => ({})` and check if it was called.

**Comment by dotansimha** [comment]
@saihaj i guess it needs a rebase?

**Comment by saihaj** [comment]
> @saihaj i guess it needs a rebase?

The failing ci is related to https://github.com/graphql/graphql-js/issues/3340

**Comment by saihaj** [comment]
@dotansimha should be good now

**Pull request merged**